scripts/ty_benchmark/src/benchmark/tool.py

```diff
@@ -0,0 +1,207 @@
+        command = [
+            str(path),
+            "--python-executable",
+            str(venv.python),
+            "--python-version",
+            project.python_version,
+            "--no-pretty",
+            *project.include,
+            "--check-untyped-defs",
+        ]
+
+        for exclude in project.exclude:
+            # Mypy uses regex...
+            # This is far from perfect, but not terrible.
+            command.extend(
+                [
+                    "--exclude",
+                    exclude.replace(".", r"\.")
+                    .replace("**", ".*")
+                    .replace("*", r"\w.*"),
+                ]
+            )
+
+        if not self.warm:
+            command.extend(
+                [
+                    "--no-incremental",
+                    "--cache-dir",
+                    os.devnull,
+                ]
+            )
+
+        return Command(
+            name="mypy (warm)" if self.warm else "mypy",
+            command=command,
+        )
+
+
+class Pyright(Tool):
+    path: Path
+
+    def __init__(self, *, path: Path | None = None):
+        if path:
+            self.path = path
+        else:
+            if sys.platform == "win32":
+                self.path = Path("./node_modules/.bin/pyright.exe").resolve()
+            else:
+                self.path = Path("./node_modules/.bin/pyright").resolve()
+
+            if not self.path.exists():
+                print(
+                    "Pyright executable not found. Did you ran `npm install` in the `ty_benchmark` directory?"
+                )
+
+    def command(self, project: Project, venv: Venv, single_threaded: bool) -> Command:
+        command = [str(self.path), "--skipunannotated"]
+
+        (venv.project_path / "pyrightconfig.json").write_text(
+            json.dumps(
+                {
+                    "exclude": [str(path) for path in project.exclude],
+                    # Set the `venv` config for pyright. Pyright only respects the `--venvpath`
+                    # CLI option when `venv` is set in the configuration... 🤷‍♂️
+                    "venv": venv.name,
+                }
+            )
+        )
+
+        if not single_threaded:
+            command.append("--threads")
+
+        command.extend(
+            [
+                "--venvpath",
+                str(
+                    venv.path.parent
+                ),  # This is not the path to the venv folder, but the folder that contains the venv...
+                "--pythonversion",
+                project.python_version,
+                "--level=warning",
+                "--project",
+                "pyrightconfig.json",
+                *project.include,
+            ]
+        )
+
+        return Command(
+            name="Pyright",
+            command=command,
+        )
+
+
+class Pyrefly(Tool):
+    path: Path
+
+    def __init__(self, *, path: Path | None = None):
+        self.path = path or which_tool("pyrefly")
```

Comment:
Hm, we depend on `pyrefly` in `pyproject.toml`, but then use whichever version we get from the PATH? If we want to pin tool versions, we should use the one from the venv?